src/ext/db/connection.cr

```diff
@@ -0,0 +1,11 @@
+module DB
+  abstract class Connection
+    getter stack = [] of DB::Transaction
+
+    def begin_transaction : Transaction
+      return stack.last if @transaction
+      @transaction = true
+      create_transaction
+    end
```

Comment:
https://github.com/crystal-lang/crystal-db/blob/5a7d27e0c58510458af3894783649d53529753cc/src/db/connection.cr#L49-L53

This needs to be thought about quite a bit. The regular implementation raises an error if it's called while there's already a transaction. Instead of doing that, I return the current transaction.

Rails' implementation returns a new nested transaction if one exists https://github.com/rails/rails/blob/b7e1ea2eeb524224dc06c08fbb779f99d20dde2f/activerecord/lib/active_record/connection_adapters/abstract/transaction.rb#L233-L263